subsys/bluetooth/audio/pacs.c

```diff
@@ -246,16 +249,39 @@ static int set_available_contexts(uint16_t contexts, uint16_t *available,
 		return 0;
 	}
 
-	*available = contexts;
-
 	err = k_work_reschedule(&available_contexts_work, PAC_NOTIFY_TIMEOUT);
 	if (err < 0) {
 		return err;
+	} else {
+		*available = contexts;
 	}
 
 	return 0;
 }
 
+static int set_supported_contexts(uint16_t contexts, uint16_t *supported,
+				  uint16_t *available)
+{
+	int err;
+
+	if (contexts & ~*available) {
+		*available = contexts;
+	}
```

Comment:
I wonder if it would be simpler if this was just
```suggestion
	/* Update available contexts if needed */
	if ((contexts & *available) != *available) {
		*available = *available & contexts;
		notify_available_contexts()
	}
```

The chance to available contexts also needs to be moved after the `k_work_reschedule` error check, otherwise we might end up with modifying the available contexts without updating the supported contexts. 

The change to available contexts also needs to be notified. 

Lastly, we cannot just assigned `contexts` to `*available`, as `*available` may still just be a subset of `contexts`

Reply:
Done